### PR TITLE
implement a* algorithm for food pathfinding

### PR DESCRIPTION
**Pull request by mattcorbin**

### CODE REVIEW

File: src/graph.rs [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+pub fn is_goal(space: Space) -> bool {
+    return space.space_type == SpaceType::Food;
+}
+
+pub fn calculate_weight(input: (Space, Space, &usize)) -> usize {
+    if input.1.space_type == SpaceType::Occupied {
+        100
+    } else {
+        0
```
Comment (jessemarion) [inline src/graph.rs:104]: Should the weight of hazard be different than food/ empty?

File: src/graph.rs [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+use crate::moves::Direction;
+use crate::{Board, Coord};
+
+#[derive(Debug, Copy, Clone, Ord, PartialOrd, Eq, PartialEq, Hash)]
+pub enum SpaceType {
+    Empty,
+    Food,
+    Hazard,
+    Occupied,
```
Comment (jessemarion) [inline src/graph.rs:13]: I think that eventually we will want to break this down by snake (especially for certain game modes)

File: src/graph.rs (hunk 1 above)
Comment (mattcorbin) [reply src/graph.rs:104]: you havent fixed your pr yet so i dont have hazard spaces in the graph

File: src/graph.rs (hunk 2 above)
Comment (mattcorbin) [reply src/graph.rs:13]: yeah but for A* it's just pathfinding between a source and destination, I think if we want to have more complicated logic here we might not just want a shortest path algorithm

**Review by jessemarion — DISMISSED** [review]

**Review by jessemarion — APPROVED** [review]

**Pull request merged**